components/board/pi/impl/servo.go

```diff
@@ -59,12 +59,22 @@ func init() {
 
 				if attr.StartPos == nil {
 					setPos := C.gpioServo(theServo.pin, C.uint(1500)) // a 1500ms pulsewidth positions the servo at 90 degrees
-					if setPos != 0 {
+					errorCode := int(setPos)
```

Comment:
please refactor into a helper function to avoid repetitive code. e.g.
```
func ConvertErrorCodeToMessage(code int, message string) error {
errorMessage, exists := picommon.PiGPIOErrorMap[errorCode]
					if exists {
						return errors.Errorf("%s: %s", message, errorMessage)
					}
					if errorCode != 0 {
						return errors.Errorf("%s: %d", message, setPos)
}
```